p2p/server.go

```diff
@@ -814,12 +819,16 @@ running:
 func (srv *Server) postHandshakeChecks(peers map[enode.ID]*Peer, inboundCount int, c *conn) error {
 	switch {
 	case !c.is(trustedConn) && len(peers) >= srv.MaxPeers:
+		dialTooManyPeers.Mark(1)
 		return DiscTooManyPeers
 	case !c.is(trustedConn) && c.is(inboundConn) && inboundCount >= srv.maxInboundConns():
+		dialTooManyPeers.Mark(1)
 		return DiscTooManyPeers
 	case peers[c.node.ID()] != nil:
+		dialAlreadyConnected.Mark(1)
 		return DiscAlreadyConnected
 	case c.node.ID() == srv.localnode.ID():
+		dialSelf.Mark(1)
 		return DiscSelf
```

Comment:
It's not great to add to these metrics here, because it means they will also be affected by incoming connections. Better to figure out the metric to add to in `setupConn` (like you are doing for the eth protocol with `markError`). Be sure to only do this for dialed connections.